app/src/main/java/com/example/placefinder/MainActivity.kt

```diff
@@ -99,13 +99,22 @@ class MainActivity : ComponentActivity() {
                             states = park.toString()
                     }) {
                         Text(text = park.toString())
-                    } 
+                    }
                     }
                 }
             }
         }
     }
 
+    @Composable
+    fun ParksList(parks: List<Address>) {
+        LazyColumn {
+            items(parks) { park ->
+                Parks(park)
+            }
+        }
+    }
+
```

Comment:
In the given code, the Parks() composable is using the Row() and Box() composable to display the parks. However, if the number of parks is large, it can lead to performance issues. Instead, you they should use the LazyColumn() composable which loads only the visible items and recycles the others.

Reply:
I don't see where this function, ParksList, is getting called, so I am not sure if this function is used... but yes, I agree.  Better to use LazyColumn than a series of Row() and Box() elements.